pkg/namespace/consts.go

```diff
@@ -0,0 +1,63 @@
+package namespace
+
+import (
+	"bytes"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+)
+
+const (
+	// NamespaveVersionSize is the size of a namespace version in bytes.
+	NamespaceVersionSize = appconsts.NamespaceVersionSize
+
+	// NamespaceIDSize is the size of a namespace ID in bytes.
+	NamespaceIDSize = appconsts.NamespaceIDSize
+
+	// NamespaceSize is the size of a namespace (version + ID) in bytes.
+	NamespaceSize = appconsts.NamespaceSize
+
+	// NamespaceVersionZero is the first namespace version.
+	NamespaceVersionZero = uint8(0)
```

Comment:
I'm not opposed. What is the benefit of using a type alias instead of using `uint8` directly?

Ref: https://yourbasic.org/golang/type-alias/ doesn't point out any strong benefits.

Reply:
Good question! For instance, if we were to allocate 16 bits for the version in the future, we could easily do so by updating the alias type. This change would immediately clarify all other places where the alias is used, making refactoring a simple and straightforward process.

Reply:
At present, the reason for defining `NamespaceVersionZero` (or any other variable holding a version type as its value) as uint8 is unclear. It is unclear whether this is intended for mathematical operations with another uint8 value or if the version is supposed to be 8 bits long by default. This question may also arise in other parts of the code.

Reply:
The three articles I read on it just now make it seem like the primary benefit is refactoring. So I agree with your point on refactoring but I think it comes at the cost of readability (subjective argument) because IMO this

```go
func NewCompactShareSplitter(ns appns.Namespace, shareVersion uint8) *CompactShareSplitter {
```

is slightly easier to read than this:

```go
func NewCompactShareSplitter(ns appns.Namespace, shareVersion appconsts.ShareVersion) *CompactShareSplitter {
```

because the latter forces you to hover or jump to another file to learn what underlying type a `ShareVersion` is. I'm a little hesitant to introduce them because:

> Type aliases are not meant for everyday use.

Ref: https://yourbasic.org/golang/type-alias/

> Type alias declaration wasn’t designed with everyday use in mind. 

Ref: https://gosolve.io/golang-type-alias/

but will open a follow-up issue if you have strong opinions about adopting it.